Locale-dependent tests failing with a French locale

We have certain tests which fail when run on a machine with a French locale set:
```
[ERROR] Failures: 
[ERROR]   ForRangeTests.testForRangeWithFloatingPointValues:82->RefineTest.parseEval:352 Wrong result for expression: forRange(0,0.6,0.1,v,v.toString('%.3f')).join(',') expected [0.000,0.100,0.200,0.300,0.400,0.500] but found [0,000,0,100,0,200,0,300,0,400,0,500]
[ERROR]   ExcelImporterTests.readXlsxAsText:284 expected [100.00%] but found [100,00%]
```

This should not be the case.

--- a/main/tests/server/src/com/google/refine/grel/controls/ForRangeTests.java
+++ b/main/tests/server/src/com/google/refine/grel/controls/ForRangeTests.java
@@ -37,7 +37,9 @@ import org.testng.annotations.Test;
 
 import com.google.refine.util.TestUtils;
 
+import java.util.Arrays;
 import java.util.Properties;
+import java.util.stream.Collectors;
 
 import static org.testng.Assert.assertTrue;
 import static org.testng.Assert.fail;
@@ -76,7 +78,14 @@ public class ForRangeTests extends RefineTest {
 
     @Test
     public void testForRangeWithFloatingPointValues() throws ParsingException {
-        String test[] = { "forRange(0,0.6,0.1,v,v.toString('%.3f')).join(',')", "0.000,0.100,0.200,0.300,0.400,0.500" };
+        // the expected result of this test is computed dynamically because it depends on the system locale
+        // (number formatting is locale-dependent)
+        String expectedResult = String.join(",",
+                Arrays.asList(0.0, 0.1, 0.2, 0.3, 0.4, 0.5).stream()
+                        .map(v -> String.format("%.3f", v))
+                        .collect(Collectors.toList()));
+
+        String test[] = { "forRange(0,0.6,0.1,v,v.toString('%.3f')).join(',')", expectedResult };
         bindings = new Properties();
         bindings.put("v", "");
         parseEval(bindings, test);

--- a/main/tests/server/src/com/google/refine/importers/ExcelImporterTests.java
+++ b/main/tests/server/src/com/google/refine/importers/ExcelImporterTests.java
@@ -281,8 +281,8 @@ public class ExcelImporterTests extends ImporterTest {
         assertEquals(project.rows.get(1).getCellValue(7), "1");
         assertEquals(project.rows.get(2).getCellValue(7), "2");
 
-        assertEquals(project.rows.get(1).getCellValue(8), "100.00%");
-        assertEquals(project.rows.get(2).getCellValue(8), "200.00%");
+        assertEquals(project.rows.get(1).getCellValue(8), String.format("%.2f", 100.0) + "%");
+        assertEquals(project.rows.get(2).getCellValue(8), String.format("%.2f", 200.0) + "%");
 
         assertEquals(project.rows.get(1).getCellValue(9), "0001");
         assertEquals(project.rows.get(2).getCellValue(9), "0002");
@@ -291,7 +291,7 @@ public class ExcelImporterTests extends ImporterTest {
 
         assertEquals(project.rows.get(ROWS - 1).getCellValue(11), NOW_STRING.substring(0, 10)); // date only
 
-        assertEquals(project.rows.get(ROWS - 1).getCellValue(12), "$1,234.56");
+        assertEquals(project.rows.get(ROWS - 1).getCellValue(12), "$" + NUMBER_FORMAT.format(1234.56));
 
         verify(options, times(1)).get("ignoreLines");
         verify(options, times(1)).get("headerLines");

